### PR TITLE
Snibblets, Tidbits, and Morsels

### PR DESCRIPTION
**Pull request by ethteck**
yummy

Fixed one of the segments (included bins and images into it, which now also includes the data and rodata that was previously bin)
Decomped a few functions
Improved coverage.py and assist.py

### CODE REVIEW

File: src/code_341d0.c [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
 
+typedef s32 TlbEntry[0x1000 / 4];
+typedef TlbEntry TlbMappablePage[15];
+
+extern TlbMappablePage D_80197000;
+extern EffectTableEntry D_8007F210[135];
+
+#define EFFECT_LOADED 1
+
+extern EffectInstance* D_800B4398[96]; //effectInstanceList
```
Comment (nanaian) [inline src/code_341d0.c:11]: rename this symbol?

Comment (nanaian) [reply src/code_341d0.c:11]: oh you moved this from another file

File: src/code_7bb60_len_41b0.c [hunk 2]
```diff
@@ -10,7 +13,40 @@ void func_800E26B0(void) {
...
+    PlayerStatus* playerStatus = gPlayerStatusPtr;
+    PlayerData* playerData = &gPlayerData;
+    s32 actionState = playerStatus->actionState;
+    s8* temp_8010EBB0 = &D_8010EBB0;
+
+    if (actionState == ActionState_IDLE ||
+        actionState == ActionState_WALK ||
+        actionState == ActionState_RUN ||
+        actionState == ActionState_USE_TWEESTER ||
+        actionState == ActionState_SPIN) {
```
Comment (nanaian) [inline src/code_7bb60_len_41b0.c:26]: can we put the `{` on a newline when the condition is long pretty please
```c
if (actionState == ActionState_IDLE ||
    actionState == ActionState_WALK ||
    actionState == ActionState_RUN ||
    actionState == ActionState_USE_TWEESTER ||
    actionState == ActionState_SPIN)
{
    return 1;
}
```
or even
```c
if (
    actionState == ActionState_IDLE ||
    actionState == ActionState_WALK ||
    actionState == ActionState_RUN ||
    actionState == ActionState_USE_TWEESTER ||
    actionState == ActionState_SPIN
) {
    return 1;
}
```

File: src/code_7bb60_len_41b0.c [hunk 3]
```diff
@@ -10,7 +13,40 @@ void func_800E26B0(void) {
...
+    if (actionState == ActionState_RIDE) {
+        if (playerData->currentPartner == PartnerID_LAKILESTER || playerData->currentPartner == PartnerID_BOW) {
+            if (temp_8010EBB0[0] != 0) {
+                return 1;
+            } else {
+                playerStatus->animFlags |= 4;
+                return 0;
+            }
+        } else {
+            if ((u32)((u8)temp_8010EBB0[3] - 6) < 2) { // or of things like the partners above
```
Comment (nanaian) [inline src/code_7bb60_len_41b0.c:39]: switch

File: src/code_7bb60_len_41b0.c [hunk 4]
```diff
@@ -37,9 +73,37 @@ INCLUDE_ASM(s32, "code_7bb60_len_41b0", update_fall_state);
...
+        playerStatus2->gravityIntegrator[1] = *floats++;
+        playerStatus2->gravityIntegrator[2] = *floats++;
+        playerStatus2->gravityIntegrator[3] = *floats++;
+    }
+}
+
+void func_800E3100(void) {
+    PlayerStatus* playerStatus = PLAYER_STATUS;
+
+    if ((playerStatus->actionState != 7) && (playerStatus->actionState != ActionState_BOUNCE)) {
```
Comment (nanaian) [inline src/code_7bb60_len_41b0.c:100]: too many parens here i think

Comment (nanaian) [reply src/code_7bb60_len_41b0.c:100]: also it's probably worth making a #define for ActionState_7 just so its easy to name later when we figure out what it is

File: src/code_7bb60_len_41b0.c [hunk 5]
```diff
@@ -69,37 +133,174 @@ f32 func_800E34D8(void) {
...
+    f32 temp_64 = playerStatus->unk_64;
+
+    if (temp_64 != 0.0f) {
+        f32 x = playerStatus->position.x;
+        f32 y = playerStatus->position.y;
+        f32 z = playerStatus->position.z;
+
+        do_lateral_collision(0, playerStatus, &x, &y, &z, temp_64, playerStatus->unk_88);
+
+        temp_64 -= (playerStatus->runSpeed / 10.0f);
```
Comment (nanaian) [inline src/code_7bb60_len_41b0.c:203]: remove parens

File: src/code_7bb60_len_41b0.c [hunk 6]
```diff
@@ -69,37 +133,174 @@ f32 func_800E34D8(void) {
...
+    f32 x = playerStatus->position.x;
+    f32 y = playerStatus->position.y;
+    f32 z = playerStatus->position.z;
+    s32 temp = func_800DF15C(&gPlayerStatus, &x, &y, &z, 0, yaw, &tempB);
+
+    playerStatus->position.x = x;
+    playerStatus->position.z = z;
 
-INCLUDE_ASM(s32, "code_7bb60_len_41b0", check_input_midair_jump);
+    if ((tempB != 0) && (temp < 0) && (playerStatus->actionState != 0x18) && (playerStatus->currentSpeed != 0.0f)) {
```
Comment (nanaian) [inline src/code_7bb60_len_41b0.c:248]: extra parens here too

File: src/code_7bb60_len_41b0.c [hunk 7]
```diff
@@ -69,37 +133,174 @@ f32 func_800E34D8(void) {
...
+        set_action_state(0x18);
+    }
+}
+
+void check_input_midair_jump(void) {
+    if (!(gPlayerStatus.flags & 0x800018) &&
+        !(gPlayerStatus.animFlags & 0x4001) &&
+        gPlayerStatus.unk_C2 >= 6 &&
+        gPlayerStatus.decorationList < 0x12 &&
+        gPlayerStatus.pressedButtons & A_BUTTON) {
```
Comment (nanaian) [inline src/code_7bb60_len_41b0.c:258]: same here with syntax of big-condition if

File: src/code_7bb60_len_41b0.c [hunk 8]
```diff
@@ -69,37 +133,174 @@ f32 func_800E34D8(void) {
...
+    }
+}
 
 PartnerID get_current_partner_id(void) {
     return gPlayerData.currentPartner;
 }
 
-INCLUDE_ASM(s32, "code_7bb60_len_41b0", func_800E5098);
+void func_800E5098(s32 arg0) {
+    if (((*gGameStatusPtr)->frameCounter % arg0) == 0) {
```
Comment (nanaian) [inline src/code_7bb60_len_41b0.c:280]: extra parens

File: src/code_7bb60_len_41b0.c [hunk 9]
```diff
@@ -111,7 +312,18 @@ INCLUDE_ASM(s32, "code_7bb60_len_41b0", can_player_interact);
...
 
-INCLUDE_ASM(s32, "code_7bb60_len_41b0", func_800E546C);
+void func_800E546C(void) {
+    f32 angle = 0.0f;
+
+    if (gPlayerStatus.spriteFacingAngle >= 90.0f && gPlayerStatus.spriteFacingAngle < 270.0f) {
+        angle = 180.0f;
+    }
+
+    // TODO LOOK
```
Comment (nanaian) [inline src/code_7bb60_len_41b0.c:322]: ?

File: src/code_7bb60_len_41b0.c [hunk 10]
```diff
@@ -111,7 +312,18 @@ INCLUDE_ASM(s32, "code_7bb60_len_41b0", can_player_interact);
...
+    angle = (angle + gCameras[0].currentYaw) + 90.0f;
```
Comment (nanaian) [inline src/code_7bb60_len_41b0.c:323]: extra parens

File: src/code_7bb60_len_41b0.c [hunk 11]
```diff
@@ -270,7 +482,42 @@ s32 check_input_hammer(void) {
...
+        !is_ability_active(Ability_SLOW_GO)) {
+
+        s8 actionState = playerStatus->actionState;
+
+        if ((playerStatus->pressedButtons & Z_TRIG) &&
+            (actionState != 0x21) &&
+            (actionState < 0x22) &&
+            (actionState < 3) &&
+            (actionState >= 0) &&
+            !(playerStatus->animFlags & 0x10000)) {
```
Comment (nanaian) [inline src/code_7bb60_len_41b0.c:503]: the ifs here too

File: src/code_7bb60_len_41b0.c [hunk 12]
```diff
@@ -270,7 +482,42 @@ s32 check_input_hammer(void) {
...
+
+            set_action_state(ActionState_SPIN);
+            if (temp_8010F250->unk_01 != 0) {
+                if (temp_8010F250->unk_08 == 0) {
+                    if (temp_8010F250->unk_0C == 0) {
+                        playerStatus->prevActionState = 0;
```
Comment (nanaian) [inline src/code_7bb60_len_41b0.c:509]: is there a #define for actionstate 0?

File: src/code_7bb60_len_41b0.c [hunk 13]
```diff
@@ -270,7 +482,42 @@ s32 check_input_hammer(void) {
...
+            (actionState != 0x21) &&
+            (actionState < 0x22) &&
+            (actionState < 3) &&
+            (actionState >= 0) &&
+            !(playerStatus->animFlags & 0x10000)) {
+
+            set_action_state(ActionState_SPIN);
+            if (temp_8010F250->unk_01 != 0) {
+                if (temp_8010F250->unk_08 == 0) {
+                    if (temp_8010F250->unk_0C == 0) {
```
Comment (nanaian) [inline src/code_7bb60_len_41b0.c:508]: does this not match with `&&`?

File: src/code_7bb60_len_41b0.c [hunk 14]
```diff
@@ -280,13 +527,76 @@ void func_800E636C(s32 arg0) {
...
+        (*gGameStatusPtr)->peachFlags &= ~0x2;
+        playerStatus->peachDisguise = 0;
+        free_npc_by_index(D_8010C96C);
+        set_action_state(ActionState_IDLE);
+        playerStatus->colliderHeight = 55;
+        playerStatus->colliderDiameter = 38;
+    }
+}
+
+Npc* make_disguise_npc(s32 peachDisguise);
```
Comment (nanaian) [inline src/code_7bb60_len_41b0.c:546]: move this to functions.h

File: src/code_7bb60_len_41b0.c [hunk 15]
```diff
@@ -280,13 +527,76 @@ void func_800E636C(s32 arg0) {
...
+extern s32 D_8010C92C;
```
Comment (nanaian) [inline src/code_7bb60_len_41b0.c:547]: move this to variables.h

**Review by nanaian — CHANGES_REQUESTED** [review]

File: src/code_341d0.c (hunk 1 above)
Comment (ethteck) [reply src/code_341d0.c:11]: I just moved all this stuff up I think

File: src/code_7bb60_len_41b0.c (hunk 2 above)
Comment (ethteck) [reply src/code_7bb60_len_41b0.c:26]: sure, I like the top one

File: src/code_7bb60_len_41b0.c (hunk 9 above)
Comment (ethteck) [reply src/code_7bb60_len_41b0.c:322]: rofl oops forgot to look

**Comment by nanaian** [comment]
Resolves #186

File: src/code_7bb60_len_41b0.c (hunk 3 above)
Comment (ethteck) [reply src/code_7bb60_len_41b0.c:39]: It's an or, not a switch. We learned that with the partnerID thing. Whenever it does a subtraction and then < some number and then an unsigned cast

Comment (ethteck) [reply src/code_7bb60_len_41b0.c:39]: I made it into an or and it works...not sure why I didn't do that before

File: src/code_7bb60_len_41b0.c (hunk 8 above)
Comment (ethteck) [reply src/code_7bb60_len_41b0.c:280]: I disagree here. Even if one knows the order of operations, I don't like having `long_expression % y == z`

File: src/code_7bb60_len_41b0.c (hunk 13 above)
Comment (ethteck) [reply src/code_7bb60_len_41b0.c:508]: There's other stuff that's wrong, so I'd rather fix the function and then do that

**Review by nanaian — APPROVED** [review]

**Pull request merged**